native/src/main/java/io/xlibb/pipe/Pipe.java

```diff
@@ -246,7 +247,7 @@ public static BError immediateClose(BObject pipe) {
     protected void asyncClose(Callback callback, long timeout) {
         if (this.isClosed.get()) {
             callback.onError(createError(CLOSED_PIPE_ERROR));
-        } else if (timeout == -1) {
+        } else if (timeout <= -1) {
```

Comment:
SHouldn't we validate this when creating the object?

Reply:
In all APIs the timeout is validated in `getTimeoutInMillis()`. 
https://github.com/xlibb/module-pipe/blob/eb374804afc9e98e19d2ba5d6d6653590b6819ba/native/src/main/java/io/xlibb/pipe/Pipe.java#L166C25-L166C43

But unlike in `consume`, `produce` APIs -1 does not considered as an infinite wait in the `gracefulClose` API. Therefore we need to separately handle only that part